### PR TITLE
Atomic improvements.

### PR DESCRIPTION
**Pull request by Nekotekina**
Reduce memory usage.
Allow notifier to skip notifications on unmet conditions.
Improve raw_notify to always wake up the thread, add thread_id arg.

### CODE REVIEW

File: rpcs3/Emu/System.cpp [hunk 1]
```diff
@@ -1700,7 +1700,11 @@ void Emulator::Run(bool start_playtime)
 	idm::check<named_thread<ppu_thread>>(ppu_thread::id_base, [](cpu_thread& cpu)
 	{
 		cpu.state -= cpu_flag::stop;
-		cpu.notify();
+
+		if (cpu.id_type() == 1)
+			thread_ctrl::raw_notify(static_cast<named_thread<ppu_thread>&>(cpu));
+		else if (cpu.id_type() == 2)
+			thread_ctrl::raw_notify(static_cast<named_thread<ppu_thread>&>(cpu));
```
Comment (vieiraa) [inline rpcs3/Emu/System.cpp:1707]:
```suggestion
		else if (cpu.id_type() == 2)
			thread_ctrl::raw_notify(static_cast<named_thread<spu_thread>&>(cpu));
```

Comment (elad335) [reply rpcs3/Emu/System.cpp:1707]: wait actually its idm::check<named_thread<ppu_thread>> so its always PPU..

Comment (elad335) [reply rpcs3/Emu/System.cpp:1707]: if `cpu` type was named_thread<ppu_thread>& there wasno need for cast or checks in the first place.

Comment (Nekotekina) [reply rpcs3/Emu/System.cpp:1707]: It' temporarily to test raw_notify, will remove.

**Comment by Nekotekina** [comment]
PR updated with new feature, possible lower CPU usage.

**Pull request merged**